### ISSUE TITLE
Tests (important??)

### ISSUE THREAD

**Issue by gauravmittal1995** [nisargjhaveri/saathi#37]
Ok, We need to start thinking about tests. Now its becoming important, because instead of reviewing everytime, we can simply leave it on the testing system (travis if we want), to do it.

@nisargjhaveri @imjalpreet Thoughts.

P.S i know this is stretching it a bit far, but i think it will make it simpler as we wont have to review each and every PR. For some, if tests are passed, then its all good.

**Comment by nisargjhaveri** [comment]
@gauravmittal1995,

Agreed that tests are extremely useful for long term maintenance of the system.

But, again tests are hand-written and if tests passes, we cannot assume PR does what it is supposed to do as the tests also are written by the author of PR.
We have to manually check each and every PR, irrespective of tests. (Yes, if tests fails, it makes our task easier :P)

Again, I am completely agreed on usefulness and importance of tests. But don't we have enough to do at this stage?

Thoughts?

**Comment by gauravmittal1995** [comment]
@nisargjhaveri Its fine by me. Lets leave this issue open for now and continue as we have. We can decide this after the first presentation
